tests/net/socket/tls/src/main.c

```diff
@@ -0,0 +1,475 @@
+
+	memset(addrstr, '\0', sizeof(addrstr));
+	addrstrp = (char *)inet_ntop(AF_INET, &sa.sin_addr,
+	addrstr, sizeof(addrstr));
+	__ASSERT(addrstrp != NULL, "inet_ntop() failed (%d)", errno);
+
+	NET_DBG("accepted connection from [%s]:%d as fd %d",
+	log_strdup(addrstr), ntohs(sa.sin_port), client_fd);
+
+	NET_DBG("calling recv()");
+	r = recv(client_fd, addrstr, sizeof(addrstr), 0);
+	__ASSERT(r != -1, "recv() failed (%d)", errno);
+	__ASSERT(r == SECRET_SIZE, "expected: %zu actual: %d", SECRET_SIZE, r);
+
+	NET_DBG("calling send()");
+	r = send(client_fd, SECRET, SECRET_SIZE, 0);
+	__ASSERT(r != -1, "send() failed (%d)", errno);
+	__ASSERT(r == SECRET_SIZE, "expected: %zu actual: %d", SECRET_SIZE, r);
+
+	NET_DBG("closing client fd");
+	r = close(client_fd);
+	__ASSERT(r != -1, "close() failed on the server fd (%d)", errno);
+}
+
+static void test_common(int peer_verify)
+{
+	const int yes = true;
+
+	int r;
+	int server_fd;
+	int client_fd;
+	int proto = IPPROTO_TCP;
+	char *addrstrp;
+	k_tid_t server_thread_id;
+	struct sockaddr_in sa;
+	char addrstr[INET_ADDRSTRLEN];
+
+	k_sem_init(&server_sem, 0, 1);
+
+	/* set the common protocol for both client and server */
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+		proto = IPPROTO_TLS_1_2;
+	}
+	/*
+	 * Server socket setup
+	 */
+
+	NET_DBG("Creating server socket");
+	r = socket(AF_INET, SOCK_STREAM, proto);
+	zassert_not_equal(r, -1, "failed to create server socket (%d)", errno);
+	server_fd = r;
+
+	r = setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes));
+	zassert_not_equal(r, -1, "failed to set SO_REUSEADDR (%d)", errno);
+
+	if (IS_ENABLED(CONFIG_TLS_CREDENTIALS)
+		&& IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+
+		static const sec_tag_t server_tag_list_verify_none[] = {
+			SERVER_CERTIFICATE_TAG,
+		};
+
+		static const sec_tag_t server_tag_list_verify[] = {
+			CA_CERTIFICATE_TAG,
+			SERVER_CERTIFICATE_TAG,
+		};
+
+		const sec_tag_t *sec_tag_list;
+		size_t sec_tag_list_size;
+
+		switch (peer_verify) {
+		case TLS_PEER_VERIFY_NONE:
+			sec_tag_list = server_tag_list_verify_none;
+			sec_tag_list_size = sizeof(server_tag_list_verify_none);
+			break;
+		case TLS_PEER_VERIFY_OPTIONAL:
+		case TLS_PEER_VERIFY_REQUIRED:
+			sec_tag_list = server_tag_list_verify;
+			sec_tag_list_size = sizeof(server_tag_list_verify);
+
+			r = setsockopt(server_fd, SOL_TLS, TLS_PEER_VERIFY,
+				&peer_verify, sizeof(peer_verify));
+			zassert_not_equal(r, -1,
+				"failed to set TLS_PEER_VERIFY (%d)", errno);
+			break;
+		default:
+			zassert_true(false,
+				"unrecognized TLS peer verify type %d",
+				peer_verify);
+			return;
+		}
+
+		r = setsockopt(server_fd, SOL_TLS, TLS_SEC_TAG_LIST,
+			sec_tag_list, sec_tag_list_size);
+		zassert_not_equal(r, -1, "failed to set TLS_SEC_TAG_LIST (%d)",
+			errno);
+
+		r = setsockopt(server_fd, SOL_TLS, TLS_HOSTNAME, "localhost",
+			sizeof("localhost"));
+		zassert_not_equal(r, -1, "failed to set TLS_HOSTNAME (%d)",
+			errno);
```

Comment:
and here. Actually I see indentation issues all over the place. Could you go through this file and fix them all in one go, I do not bother comment them all.